packages/bvaughn-architecture-demo/components/console/ConsoleRoot.tsx

```diff
@@ -23,24 +23,39 @@ import { LoggablesContextRoot } from "./LoggablesContext";
 import MessagesList from "./MessagesList";
 import Search from "./Search";
 import { SearchContextRoot } from "./SearchContext";
+import { SessionContext } from "@bvaughn/src/contexts/SessionContext";
 
 export default function ConsoleRoot({
   nagHeader = null,
   showSearchInputByDefault = true,
+  showFiltersByDefault = false,
   terminalInput = null,
 }: {
   filterDrawerOpenDefault?: boolean;
   nagHeader?: ReactNode;
   showSearchInputByDefault?: boolean;
+  showFiltersByDefault?: boolean;
   terminalInput?: ReactNode;
 }) {
   const { clearMessages: clearConsoleEvaluations, messages: consoleEvaluations } =
     useContext(TerminalContext);
 
-  const [isMenuOpen, setIsMenuOpen] = useLocalStorage<boolean>("Replay:Console:MenuOpen", true);
+  const { recordingId } = useContext(SessionContext);
+  const [isMenuOpen, setIsMenuOpen] = useLocalStorage<boolean>(
+    `Replay:Console:MenuOpen:${recordingId}`,
+    showFiltersByDefault
+  );
+  const [menuValueHasBeenToggled, setMenuValueHasBeenToggled] = useState(false);
```

Comment:
I would argue that you could change this to `const [filtersHaveBeenShown, setFiltersHaveBeenShown] = useState(showFiltersByDefault)` and skip the `||` down on line 57.

Reply:
Eh, actually, this is fine 🙂  I considered proposing this change:

```
  const filtersHaveBeenRendered: boolean = useMemo(() => {
    return !!(isMenuOpen || filtersHaveBeenRendered);
  }, [isMenuOpen]);
```

Just because I feel like the `useState` hook introduces more cognitive load than the `useMemo` hook, but I think it's mostly just a matter of taste, and that's not a great reason to go around changing working code!